packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -489,6 +499,15 @@ export class Function extends FunctionBase {
       managedPolicies.push(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaVPCAccessExecutionRole'));
     }
 
+    if (props.filesystems) {
+      for (const fs of props.filesystems) {
+        // add additonal managed policies when necessary
+        if (fs.config.managedPolicies) {
```

Comment:
what happens if you add two EFS files systems? Seems like there is going to be a duplicate managed policy entry

Reply:
should be fixed. Probably not a smart fix but it works.